### DISCUSSION TITLE
Tinygrad on Coral TPU's ?

### DISCUSSION THREAD

**Discussion by metacritical** [tinygrad/tinygrad#752] [Ideas]
Any plan of running on Coral TPU's https://coral.ai/products/#production-products ?

**Comment by geohot** [comment]
It's a weird custom undocumented arch. Not really worth spending time, but I did reverse a good bit of it. https://github.com/geohot/edgetpuxray

Also, like many Google things, it seems pretty abandoned. No update to the chip, and no real updates to the repos since 2020. https://github.com/google-coral/edgetpu/blob/master/CHANGES.md

**Reply by metacritical** [reply]
This is neat, thanks for pointing to the compatibility link.

**Reply by tomeuv** [reply]
@geohot Sorry to revive this old thread, but I'm curious about what is your stance regarding support in tinygrad for compute/systolic array NPUs/TPUs/NNAs, etc

Is the plan to only support programmable units such as CPUs and GPUs?
